pkg/infra/game/stage/fight/fight.go

```diff
@@ -100,9 +107,9 @@ func HandleFightRound(state state.State, agents map[commons.ID]agent.Agent, base
 		state.AgentState[agentID] = agentState
 	}
 
-	return coweringAgents, attackSum, shieldSum, decisionMap
+	return coweringAgents, attackSum, shieldSum
 }
 
-func startAgentFightHandlers(view state.View, a *agent.Agent, decisionLog immutable.Map[commons.ID, decision.FightAction], channel chan message.ActionMessage, wg *sync.WaitGroup) {
-	go a.HandleFight(view, decisionLog, channel, wg)
+func startAgentFightHandlers(view state.View, a *agent.Agent, decisionLog *immutable.Map[commons.ID, decision.FightAction], channel chan message.ActionMessage, wg *sync.WaitGroup) {
```

Comment:
Why is this now a pointer to decisionLog? I think this will become racy since unsafe stuff can happen between goroutines, i.e. writing directly to the same pointer.